src/lib/storage/table.cpp

```diff
@@ -67,10 +72,37 @@ const std::string& Table::column_name(ColumnID column_id) const { return _column
 
 const std::string& Table::column_type(ColumnID column_id) const { return _column_types.at(column_id); }
 
-Chunk& Table::get_chunk(ChunkID chunk_id) { return _chunks.at(chunk_id); }
+Chunk& Table::get_chunk(ChunkID chunk_id) {
+  std::lock_guard<std::mutex> lock_guard(*_access_mutex);
+  return _chunks.at(chunk_id);
+}
 
 const Chunk& Table::get_chunk(ChunkID chunk_id) const { return _chunks.at(chunk_id); }
 
-void Table::compress_chunk(ChunkID chunk_id) { throw std::runtime_error("Implement Table::compress_chunk"); }
+void Table::compress_chunk(ChunkID chunk_id) {
```

Comment:
In your Multithreading approach (if we understand it correctly) we are wondering whether it is necessary to have this many `lock_guards`. Since we are only reading from the old data structure and only inserting in the new, one lock guard for writing into the new structure should be sufficient. With adding more there is always the risk of running into deadlock situations and related issues.